packages/truffle-debugger/lib/data/sagas/index.js

```diff
@@ -72,6 +80,25 @@ export function* decode(definition, ref) {
         //any storage it does not know is presumed to be zero.
         response = ZERO_WORD;
         break;
+      case "code":
+        let address = request.address;
+        if (address in instances) {
+          response = instances[address];
+        } else if (address === DecodeUtils.EVM.ZERO_ADDRESS) {
+          //HACK: to avoid displaying the zero address to the user as an
+          //affected address just because they decoded a contract or external
+          //function variable that hadn't been initialized yet, we give the
+          //zero address's codelessness its own private cache :P
+          response = NO_CODE;
+        } else {
+          //I don't want to write a new web3 saga, so let's just use
+          //obtainBinaries with a one-element array
+          debug("fetching binary");
+          let binary = (yield* web3.obtainBinaries([address], blockNumber))[0];
+          debug("adding instance");
+          yield* evm.addInstance(address, binary);
+          response = DecodeUtils.Conversion.toBytes(binary);
+        }
```

Comment:
missing `break;`

Reply:
Ooh, missed that, thanks

Reply:
Also of course it's immediately useful, this is *why* we switched to that, remember? :P

Reply:
but it's still really neat!